# Import Data

In [1]:
library(SDSFoundations,lib.loc = 'C:\\Program Files\\R\\R-3.6.1\\library')
acl <- AustinCityLimits

In [2]:
nrow(acl)

[1] 116

In [3]:
str(acl)

'data.frame':	116 obs. of  14 variables:
 $ Artist       : Factor w/ 116 levels "Aimee Mann","Alabama Shakes",..: 1 2 3 4 5 6 7 8 9 10 ...
 $ Year         : int  2008 2013 2009 2009 2007 2009 2010 2009 2003 2008 ...
 $ Month        : Factor w/ 6 levels "December","February",..: 4 2 3 5 4 4 3 4 3 5 ...
 $ Season       : Factor w/ 2 levels "fall","winter": 1 2 2 1 1 1 2 1 2 1 ...
 $ Gender       : Factor w/ 2 levels "F","M": 1 1 2 2 1 2 2 2 2 1 ...
 $ Age          : int  52 24 75 39 33 62 37 35 43 67 ...
 $ Age.Group    : Factor w/ 4 levels "Fifties or Older",..: 1 4 1 3 3 1 3 3 2 1 ...
 $ Grammy       : Factor w/ 2 levels "N","Y": 2 1 1 1 2 2 1 1 2 1 ...
 $ Genre        : Factor w/ 4 levels "Country","Jazz/Blues",..: 4 3 2 3 3 1 3 3 3 2 ...
 $ BB.wk.top10  : int  0 1 NA 1 1 0 1 NA 1 0 ...
 $ Twitter      : int  101870 73313 308634 56343 404439 3326 125758 8197 158647 690 ...
 $ Twitter.100k : int  1 0 1 0 1 0 1 0 1 0 ...
 $ Facebook     : int  113576 298278 10721 318313 1711685 27321 56

# Question 1
#### You want to see if an artist's popularity on Facebook (whether or not they have 100,000 or more likes) has anything to do with their age.

Generate a table to show the number of artists that are "popular" and those that are not.

In [4]:
table(acl$Facebook.100k)


 0  1 
31 85 

 Generate a table to show the number of "popular" artists within each age group.

In [6]:
ctab <- table(acl$Facebook.100k,acl$Age.Group)
ctab

   
    Fifties or Older Forties Thirties Twenties
  0               13       6        9        3
  1               19      24       31       11

In [10]:
round(prop.table(ctab),3)

   
    Fifties or Older Forties Thirties Twenties
  0            0.112   0.052    0.078    0.026
  1            0.164   0.207    0.267    0.095

In [9]:
round(prop.table(ctab,1),3)

   
    Fifties or Older Forties Thirties Twenties
  0            0.419   0.194    0.290    0.097
  1            0.224   0.282    0.365    0.129

For each age group, fill in the proportion of artists who have 100,000 or more likes on Facebook. (Use the appropriate function in R to calculate these, and round to 3 decimal places (i.e. 0.123.)

In [11]:
cfblikes <- table(acl$Facebook.100k,acl$Age.Group)

In [12]:
cfblikes

   
    Fifties or Older Forties Thirties Twenties
  0               13       6        9        3
  1               19      24       31       11

In [14]:
round(prop.table(cfblikes,2),3)

   
    Fifties or Older Forties Thirties Twenties
  0            0.406   0.200    0.225    0.214
  1            0.594   0.800    0.775    0.786

# Question 2
#### A high school counselor wants to categorize students according to two variables: their gender (male or female) and their grade level (freshman, sophomore, junior or senior).

To how many possible outcomes can students be assigned?

- 4

#### An Economics professor kept a record of grades earned by the college students in his class, as shown below. (In the US, an "A" is the strongest grade and an "F" is the Weakest.)

||A|B|C|D|F|
|-|-|-|-|-|-|
|Freshman|5|10|9|10|6|
|Sophomore|8|10|9|7|4|
|Junior|11|5|4|4|2|
|Senior|9|9|4|2|0|

What proportion of students in the class received a grade of A? (Round to 2 decimal places.)

In [29]:
level <- c('Freshman','Sophomore','Junior','Senior','Total')
A <- c(5,8,11,9)
A <- c(A,sum(A))
B <- c(10,10,5,9)
B <- c(B,sum(B))
C <- c(9,9,4,4)
C <- c(C,sum(C))
D <- c(10,7,4,2)
D <- c(D,sum(D))
F <- c(6,4,2,0)
F <- c(F,sum(F))
Total <- c()
for (t in 1:5){
    Total <- c(Total,A[t]+B[t]+C[t]+D[t]+F[t])
}
df <- data.frame(level,A,B,C,D,F,Total)

In [30]:
df

level     A  B  C  D  F  Total
1 Freshman   5 10  9 10  6  40  
2 Sophomore  8 10  9  7  4  38  
3 Junior    11  5  4  4  2  26  
4 Senior     9  9  4  2  0  24  
5 Total     33 34 26 23 12 128

In [32]:
prop.table(df[5,c('A','B','C','D','F')])

A         B        C        D         F      
5 0.2578125 0.265625 0.203125 0.1796875 0.09375

- grade A recievers : 0.258

What proportion of the students were upperclassmen (juniors and seniors)? (Round to 2 decimal places.)

In [40]:
sum(df[c(3,4),7])/df[5,7]

[1] 0.390625

What is the probability that a freshman received a failing grade of F? (Round to 2 decimal places.)

In [41]:
df[1,2:6]

A B  C D  F
1 5 10 9 10 6

In [42]:
prop.table(df[1,2:6])

A     B    C     D    F   
1 0.125 0.25 0.225 0.25 0.15

What is the probability that a randomly selected student from the class would be a sophomore that received a grade of B? (Round to 2 decimal places.)

In [44]:
round(prop.table(df[1:4,2:6]),2) # marginal distribution

A    B    C    D    F   
1 0.04 0.08 0.07 0.08 0.05
2 0.06 0.08 0.07 0.05 0.03
3 0.09 0.04 0.03 0.03 0.02
4 0.07 0.07 0.03 0.02 0.00

What proportion of juniors passed the course with a grade of D or better? (Round to 2 decimal places.)

In [47]:
round(sum(df[5,2:5])/df[5,7],2)

[1] 0.91

What is the probability that a randomly selected student from this class would be a senior? (Report to 2 decimal places.)

In [49]:
round(df[4,7]/df[5,7],2)

[1] 0.19

If a student received a grade of D in the class, what is the probability that the student was a senior? (Round to 2 decimal places.)

In [50]:
round(prop.table(df[1:4,5]),2)

[1] 0.43 0.30 0.17 0.09

# Question 3
#### Use the below probability statements to answer the following two questions. Report answers as proportions.
- P(A) = 0.35
- P(A and B) = 0.15

If A and B are independent, what is the value of P(A|B)? (Round to 2 decimal places.)

0.35

What is the probability of P(B|A)? (Round to 2 decimal places.)

In [53]:
round(0.15/0.35,2)

[1] 0.43

# Question 4
#### A movie theater conducted a survey to determine the movie preferences of men and women. They asked a total of 130 adults (50 women and 80 men) to choose their favorite movie genre out of four choices: Action, Comedy, Horror, or Romance. The results of their survey are shown below.

||Women|Men|
|-|-|-|
|Action|12%|35%|
|Comedy|34%|25%|
|Horror|30%|32.50%|
|Romance|24%|7.50%|

In [58]:
women <- c((12/100)*50,(34/100)*50,(30/100)*50,(24/100)*50)
men <- c((35/100)*80,(25/100)*80,(32.50/100)*80,(7.50/100)*80)
df <- data.frame(women,men)
df

women men
1  6    28 
2 17    20 
3 15    26 
4 12     6

What is the probability that a randomly chosen person from the survey prefers Action films? (Report as a proportion rounded to 2 decimal places.)

In [60]:
round(sum(df[1,])/130,2)

[1] 0.26

What is **P(Action|Women)**? (Report as a proportion rounded to 2 decimal places.)

In [61]:
(6/130)/(50/130)

[1] 0.12

In [62]:
50/130

[1] 0.3846154

In [63]:
6/50

[1] 0.12